src/org/rascalmpl/uri/URIUtil.java

```diff
@@ -55,8 +56,14 @@ public static URI create(String scheme, String userInformation, String host, int
 	 * @throws URISyntaxException
 	 */
 	public static URI createFile(String path) throws URISyntaxException {
-		path = fixWindowsPath(path);
-		return fixUnicode(new File(path).toURI());
+		File file = new File(fixWindowsPath(path));
+		try {
+			file = file.getCanonicalFile();
```

Comment:
Here the semantics of SourceLocations becomes much much more dependent on how URIUtil works. It is very urgent to move this behavior into the IValueFactory such that nobody can forget to use it anymore. Also that would provide an opportunity for other normalization behavior that is needed for the communication with VScode (lowercasing the schemes and the authorities).

Reply:
To be fair, this function is hard to hit from user code. We only call this when we are setting up paths ourselves. 

Also it it would be slow to do this for all file paths in vallang, since it actually goes to the file system for some lookups. We don't want that all the time.